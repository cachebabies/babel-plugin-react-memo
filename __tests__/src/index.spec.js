import * as babel from 'babel-core';
import plugin from '../../src';

describe('index', () => {
  describe('ImportDeclaration', () => {
    it('should add named export to react default export', () => {
      const program = 'import React from \'react\'';
      const { code } = babel.transform(program, { plugins: [plugin] });
      expect(code).toMatchSnapshot();
    });

    it('should extend named exports', () => {
      const program = 'import React, { Component, Foo } from \'react\'';
      const { code } = babel.transform(program, { plugins: [plugin] });
      expect(code).toMatchSnapshot();
    });

    it('should not extend named exports if not react', () => {
      const program = 'import { isEqual } from \'lodash\'';
      const { code } = babel.transform(program, { plugins: [plugin] });
      expect(code).toMatchSnapshot();
    });
  });

  describe('ArrowFunctionExpression', () => {
    it('should wrap arrow function with memo', () => {
      const program = 'const fakeComponent = () => { return (<div>hello</div>); }';
      const { code } = babel.transform(program, {
        plugins: [plugin, 'syntax-jsx'],
      });
      expect(code).toMatchSnapshot();
    });

    it('should not wrap arrow function with memo within ClassProperty', () => {
      const program = 'class fakeComponent extends Component { jsxreturn() { return <div>hello</div> } render() { return (<div>hello</div>); } }';
      const { code } = babel.transform(program, {
        plugins: [plugin, 'syntax-jsx'],
      });
      expect(code).toMatchSnapshot();
    });

    it('should not wrap arrow function with memo within ClassBody', () => {
      const program = 'class fakeComponent extends Component { render() { const jsxreturn = () => { return (<div>hello</div>); } } }';
      const { code } = babel.transform(program, {
        plugins: [plugin, 'syntax-jsx'],
      });
      expect(code).toMatchSnapshot();
    });
  });
});
