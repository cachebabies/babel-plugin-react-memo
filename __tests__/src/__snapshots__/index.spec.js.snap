// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index ArrowFunctionExpression should not wrap arrow function with memo within ClassBody 1`] = `
"class fakeComponent extends Component {
  render() {
    const jsxreturn = () => {
      return <div>hello</div>;
    };
  }
}"
`;

exports[`index ArrowFunctionExpression should not wrap arrow function with memo within ClassProperty 1`] = `
"class fakeComponent extends Component {
  jsxreturn() {
    return <div>hello</div>;
  }render() {
    return <div>hello</div>;
  }
}"
`;

exports[`index ArrowFunctionExpression should not wrap function that is already wrapped 1`] = `
"const fakeComponent = memo(() => {
        return <div>hello</div>;
});
const fakeFunction = () => null;"
`;

exports[`index ArrowFunctionExpression should only wrap valid JSX component 1`] = `
"const fakeComponent = memo(() => {
        return <div>hello</div>;
});
const fakeFunction = () => null;"
`;

exports[`index ArrowFunctionExpression should wrap arrow function with memo 1`] = `
"const fakeComponent = memo(() => {
  return <div>hello</div>;
});"
`;

exports[`index ArrowFunctionExpression should wrap function that returns jsx directly 1`] = `"const fakeComponent = memo(() => <div>hello</div>);"`;

exports[`index ImportDeclaration should add named export to react default export 1`] = `"import React, { memo } from 'react';"`;

exports[`index ImportDeclaration should extend named exports 1`] = `"import React, { Component, Foo, memo } from 'react';"`;

exports[`index ImportDeclaration should not extend named exports if not react 1`] = `"import { isEqual } from 'lodash';"`;
